### ISSUE TITLE
Nexus-prisma is not ready for Typescript strict mode

### ISSUE THREAD

**Issue by nagman** [graphql-nexus/nexus-plugin-prisma#350]
Hi,

I've recently switched my typescript project to `strict` mode because I didn't understand why vs code wasn't warning me about accessing properties of potentially null or undefined objects.

Then the nightmare happened...

I had to tweak a lot of my resolvers (and even add some) because of many type errors coming from the generated client.

For instance:
```
export const UserMutations = prismaExtendType({
  type: 'Mutation',
  definition(t) {
    t.field('createUser', {
      ...t.prismaType.createUser,
      resolve(root, args, ctx) {
        return t.prismaType.createUser.resolve(
          root,
          { data: { ...args.data, password: '' } },
          ctx,
        );
      },
    });
  },
});
```

I want to set the password to an empty string when creating a User.

Before `strict` mode, I had no error with the linter (except [this one](https://github.com/prisma/nexus-prisma/issues/291) for which I had to add `// @ts-ignore` before the `export`).

But since I've switched `strict` to `true` in `tsconfig.json`, I have this (summarized) error:
```
Types of property 'clients' are incompatible.
  Types of property 'connect' are incompatible.
    Type '{ id?: string | null | undefined; }[] | null | undefined' is not assignable to type 'ClientWhereUniqueInput[] | undefined'.
      Type 'null' is not assignable to type 'ClientWhereUniqueInput[] | undefined'.
```
In my datamodel, a User can have Clients.

But I don't understand why the generated createUser args could be null and the generated createUser resolve function doesn't accept null.

How does the default resolvers interact with the args?  
I mean, if I can't use directly the generated args with the generated resolve function, how nexus-prisma would have handled the `createUser` endpoint?  
Or maybe nexus-prisma is simply not using strict mode.

How would you hook a generated resolver to add or remove args, if not?

I hope I'm clear enough.

**Comment by jasonkuhrt** [comment]
Hey @nagman, just wanted to let you know that this issue is resolved with Prisma 2 (`nexus-prisma` >= 0.4). You can see this for yourself [in the example](https://github.com/prisma-labs/nexus-prisma/blob/master/examples/blog/src/schema/Query.ts#L11-L44).

**Comment by Weakky** [comment]
I think we can close this now!
